### ISSUE TITLE
Project Feedback

### ISSUE THREAD

**Issue by KatieMFritz** [KatieMFritz/MI-000-SS00-000-js-firebase-intro-T4LtxO#1]
@egillespie Can you take a look at this? It's [hosted here](https://katie-kgucykegmdzxwt4ltxo.surge.sh/) and meets the following criteria:

- [x] Use [this code](https://gist.github.com/egillespie/942cc3aed19977590b431c10333736c2) as a foundation for your project
- [x] Use the `TODO` comments in `woofer-db.js` as your guide for placing code
- [x] Your own Firebase project is used to store all of the site's messages
- [x] A message and its creation time are added to the database when Enter is pressed or Woof is clicked
- [x] Messages are stored in the database according to the structure in `model.json`
- [x] Anyone can edit any message by clicking the pencil icon, typing a new message, and pressing Enter
- [x] Anyone can delete any message by clicking the red X icon
- [x] When a message is added, edited, or deleted, _Recent woofs_ automatically updates for everyone viewing the page

It was a lot easier with the restructured files - a good mix of figuring out what to do while still having guidance.

**Comment by egillespie** [comment]
This is great! It all works and looks great. I have a couple of recommendations to help you tighten up the code if you want:

### Push `woof` when creating record

Instead of creating a new object to create a woof in Firebase, you can pass `woof` to save some typing (and a tiny bit of execution) time:

```js
firebase.database().ref('woofs').push(woof)
```

### Referencing a field to update it

An alternative way to update a single field is to get a reference to that field and set its value:

```js
  firebase.database().ref('woofs').child(woofKey).child('text').set(woofText)
```

When passing a single field to `set` when the reference is a whole record can be confusing because it's not clear until you've tried it that Firebase won't remove the other fields.

---

Fantastic work on your first Firebase project! What do you think of it?

**Comment by KatieMFritz** [comment]
Thanks for the tips! 

I'm not sure what you mean here:
> When passing a single field to set when the reference is a whole record can be confusing because it's not clear until you've tried it that Firebase won't remove the other fields.

Is it confusing the way I did it, or the way you just recommended?

I really like Firebase and I'm thinking of at least 2 side projects I want to use it for. 😁

**Comment by egillespie** [comment]
It's not confusing to me the way you did it because I've used Firebase enough to know that

```js
firebase.database().ref('woofs').child(woofKey).set({
  text: woofText
})
```

is not intentionally trying to remove the `created_at` field. When I was first learning Firebase and saw this approach to update a single field, my first question was "It looks like this is updating the whole record. Does that mean it's going to remove the fields not listed?"

Now I know that it won't remove un-listed fields, but I developed a preference for trying to be as specific as possible with my updates so it's clear to people reading my code who aren't as familiar with Firebase.
